node/external/transactionAPI/apiTransactionProcessor.go

```diff
@@ -86,6 +87,48 @@ func NewAPITransactionProcessor(args *ArgAPITransactionProcessor) (*apiTransacti
 	}, nil
 }
 
+func (atp *apiTransactionProcessor) GetSCRsByTxHash(txHash string, scrHash string) ([]*transaction.ApiSmartContractResult, error) {
+	decodedScrHash, err := hex.DecodeString(scrHash)
+	if err != nil {
+		return nil, err
+	}
+
+	decodedTxHash, err := hex.DecodeString(txHash)
+	if err != nil {
+		return nil, err
+	}
+
+	if !atp.historyRepository.IsEnabled() {
+		return []*transaction.ApiSmartContractResult{}, nil
```

Comment:
I would return error here to signal the fact that this functionality was requested on a regular observer/validator and it can not be reliably computed by such a node

Reply:
changed